docs/user/metal/install_ipi.md

```diff
@@ -195,3 +195,80 @@ network configuration, but fully supporting alternative subnets for the
 provisioning network is incomplete.
 
 https://github.com/openshift/installer/issues/2091
+
+## Troubleshooting
+
+General troubleshooting for OpenShift installations can be found
+[here](/docs/user/troubleshooting.md).
+
+### Bootstrap
+
+The bootstrap host is currently a libvirt VM running on the same host as
+the installer. This bootstrap host runs the Ironic services needed to
+provision the control plane. Ironic being available is dependent on
+having successfully downloaded the RHCOS and Ironic agent images. In
+some cases, this may fail, and the installer will report a timeout
+waiting for the Ironic API.
+
+You can view the Ironic logs by sshing to the bootstrap node, and
+examining the logs of the `ironic` service, `journalctl -u ironic`. You
+may also view the logs of the individual containers:
+
+  - `podman logs ipa-downloader`
+  - `podman logs coreos-downloader`
+  - `podman logs ironic`
+  - `podman logs ironic-inspector`
+  - `podman logs ironic-dnsmasq`
+
+To login to the bootstrap VM, you will need to ssh to the host using the
+`core` user, and the SSH key defined in your install config.
+
+The VM obtains an IP address from your DHCP server on the external
+network. When using a development environment with
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts), it uses
+the `baremetal` libvirt network and the IP can be retrieved with `virsh
+net-dhcp-leases baremetal`. If the install is far enough along to have
+brought up the provisioning network, you may use the provisioning
+bootstrap IP which defaults to 172.22.0.2.
+
+Viewing the virtual machine's console with virt-manager may also be
+helpful.
+
+### Control Plane
+
+Once Ironic is available, the installer will provision the three control
+plane hosts. For early failures, it may be useful to look at the console
+(using virt-manager if emulating baremetal with vbmc, or through the BMC
+like iDRAC) and see if there are any errors reported.
+
+Additionally, if the cluster comes up enough that the bootstrap is destroyed,
+but commands like `oc get clusteroperators` shows degraded operators, it
+may be useful to examine the logs of the pods within the
+`openshift-kni-infra` namespace.
+
+### Ironic
+
+You may want to examine Ironic itself and look at the state of the
+hosts. The below file, when named clouds.yaml and placed in the current
+working directory, can be used to communicate with Ironic using the
+openstack commandline utilities.
```

Comment:
We should probably reference some ways to install this client, or else revive https://github.com/openshift/ironic-image/pull/17

IMO it'd probably be better to show a way to run it via a container, since we know not all downstream consumers of the installer will have access to repos with openstack RPMs, and they probably won't want to pip install the clients?

Reply:
Could we defer this until later until we have a better mechanism? I think distributing it in a container is a good idea.

Reply:
I think deferring and iterating is fine, so LGTM.

I'm actually wondering how much info we want to include about getting to ironic. I would consider it a bug if a user had to connect to ironic to understand what's going on in the installation process. I would expect the installer to report the details that might be useful in its logs.

Reply:
Definitely agree. https://github.com/openshift/installer/issues/2679, https://github.com/openshift/installer/issues/2009

Reply:
Ack agreed, lets defer it - maybe we can work out part of the gather workflow that retrieves the ironic data needed for debugging